core/src/sigverify_stage.rs

```diff
@@ -248,20 +248,27 @@ impl SigVerifyStage {
             num_packets,
         );
 
-        let mut dedup_time = Measure::start("sigverify_dedup_time");
-        let dedup_fail = deduper.dedup_packets(&mut batches) as usize;
-        dedup_time.stop();
-        let num_unique = num_packets.saturating_sub(dedup_fail);
+        use rand::{thread_rng, Rng};
+        const MAX_BATCHES: usize = 2000;
+        if batches.len() > MAX_BATCHES {
+            let percent_to_keep = (batches.len() - MAX_BATCHES) / batches.len();
+            batches.retain(|_batch| thread_rng().gen_ratio(percent_to_keep as u32, 100));
+        }
 
+        let mut num_valid_packets = num_packets;
         let mut discard_time = Measure::start("sigverify_discard_time");
-        let mut num_valid_packets = num_unique;
-        if num_unique > MAX_SIGVERIFY_BATCH {
+        if num_packets > MAX_SIGVERIFY_BATCH {
```

Comment:
Hmm, I'd have expected that we keep this discard after dedup.

Otherwise we feed only up to 10k packets into dedup and if we get 95% dups, very few would make it to verify.

Reply:
Yea. we could do that